### ISSUE TITLE
XmlSerializer corrupts empty tag when serialized using StreamWriter

### ISSUE THREAD

**Issue by Falco20019** [dotnet/runtime#46974]
### Description

Serializing an empty string/tag embedded in an `XmlElement` using `XmlSerializer` with indentation and line breaks will corrupt the data.

I need to encapsulate part of an input `XmlElement` into another class instance with additional data.

Input:
```xml
<?xml version="1.0" encoding="utf-8"?>
<CreateTransportRequest>
  <transportId>42</transportId>
  <v1></v1>
</CreateTransportRequest>
```

Expected output:
```xml
<?xml version="1.0" encoding="utf-8"?>
<Request xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <SomeStuff>Lorem Ipsum</SomeStuff>
  <Payload>
    <CreateTransportRequest>
      <transportId>42</transportId>
      <v1 />
    </CreateTransportRequest>
  </Payload>
</Request>
```

Received output:
```xml
﻿<?xml version="1.0" encoding="utf-8"?>
<Request xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <SomeStuff>Lorem Ipsum</SomeStuff>
  <Payload>
    <CreateTransportRequest>
      <transportId>42</transportId>
      <v1>
      </v1>
    </CreateTransportRequest>
  </Payload>
</Request>
```

The `v1` field will contain a line break and therefore data. I would be fine with `<v1 />` or `<v1></v1>` but it MUST NOT contain any data after serialization.

Full example:
https://dotnetfiddle.net/Pe8KDj

### Configuration

Tested with .NET 4.7.2, .NET Core 3.1 and .NET 5.0 on Windows 10 v1909. Same behavior on all version.
Also tested with .NET Fiddle on .NET 4.7.2 and .NET 5.0 for the same results (see link above).

### Other information

Using `using (var sw = XmlWriter.Create(ms, new XmlWriterSettings { Indent = true, Encoding = Encoding.UTF8 }))` instead of `using (var sw = new StreamWriter(ms, Encoding.UTF8))` fixes the problem. Still, it feels more like a workaround.

**Comment by mayorovp** [comment]
MCVE:

```cs
    public class Foo
    {
        [XmlAnyElement] // can be reproduced with XmlElement too
        public XmlElement Payload { get; set; }
    }

    class Program
    {
        static void Main(string[] args)
        {
            var doc = new XmlDocument();
            var payload = doc.CreateElement("Payload");
            payload.InnerXml = "<bar></bar>";

            var sw = new StringWriter();
            new XmlSerializer(typeof(Foo)).Serialize(sw, new Foo { Payload = payload });
            Console.WriteLine(sw);
        }
    }
```

**Comment by StephenMolloy** [comment]
I believe this was fixed with #54949
